src/App.tsx

```diff
@@ -1,61 +1,102 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TODO } from './types/todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos] = useState([...todosFromServer]);
+
+  const [title, setTitle] = useState('');
+  const [currUser, setCurrUser] = useState(0);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setCurrUser(0);
+  };
+
+  const submit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (currUser === 0) {
+      setUserError(true);
+    }
+
+    if (!title || !currUser) {
+      return;
+    }
+
+    const todo: TODO = {
+      id: [...visibleTodos].sort((a, b) => b.id - a.id)[0].id + 1 || 1,
+      title,
+      userId: currUser,
+      completed: false,
+    };
+
+    setVisibleTodos([...visibleTodos, todo]);
+    reset();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={e => submit(e)} action="/api/todos" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            type="text"
+            id="title"
+            data-cy="titleInput"
+            value={title}
+            placeholder="Enter a title"
+            onChange={e => {
+              setTitle(e.target.value);
+              setTitleError(false);
+            }}
+          />
+          {titleError && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
+          <label htmlFor="user">User: </label>
+          <select
+            value={currUser}
+            id="user"
+            onChange={e => {
+              setCurrUser(parseInt(e.target.value));
+              setUserError(false);
+            }}
+            data-cy="userSelect"
+          >
+            <option value={0} disabled>
               Choose a user
             </option>
+            {usersFromServer.map((user, index) => (
+              <option key={index} value={user.id}>
```

Comment:
Using `index` as the `key` for list items can lead to issues. It's better to use a unique identifier like `user.id` for the `key` prop to ensure stable identity across re-renders.